### ISSUE TITLE
Add ScalaGenerator support for Oracle TYPE .. UNDER POJO hierarchies

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#17189]
jOOQ 3.20 adds support for Oracle style `TYPE .. UNDER` hierarchies:

- https://github.com/jOOQ/jOOQ/issues/644

One missing feature so far is `ScalaGenerator` support for POJO hierarchies, as this doesn't work:

```scala
class U31(
  var a: Integer
)

class U32(
  override var a: Integer,
  var b: Integer
) extends U31
```

Resulting in:

> Mutable variable cannot be overridden

Perhaps there's a workaround

**Comment by lukaseder** [comment]
This works:

```scala
abstract class A (var s: String)
case class B (
  s0: String,  // No var, alternative name
  var t: String
) extends A(s0)
```

But the constructor parameter cannot reuse the name of the property, which is also not very nice

**Comment by lukaseder** [comment]
Another workaround suggested here:
https://x.com/domdorn/status/1831703991043981446

Very ugly, but it would achieve what I'm after:

```scala
class A(private var s1: String, dummy: AnyRef) { 
  
  def this(s: String) = { 
    this(s, null)
  } 

  def s = { s1 } 
  def s_=(s: String) = { s1 = s } 
}

class B(private var s2: String, private var t2: String, private var dummy: AnyRef) extends A(s2) { 
  def this(s: String, t: String) = {
    this(s, t, null)
  }

  def t = { t2 }
  def t_=(t: String) = { t2 = t }

}


val aSample = new A(s = "test")

val bSample = new B(s = "test", t = "test2")

aSample.s = "testS"
println(aSample.s)

bSample.s = "testS"
bSample.t = "testT"

println(bSample.s)
println(bSample.t)
```

**Comment by lukaseder** [comment]
I made a mistake, previously:

> ```scala
> abstract class A (var s: String)
> case class B (
>   s0: String,  // No var, alternative name
>   var t: String
> ) extends A(s0)
> ```

I didn't intend for `B` to be a `case class`, but just an ordinary `class`. This works just fine!

```scala
class A (var s: String)
class B (
  s: String, // No var
  var t: String
) extends A(s0)
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.20.0
